src/pages/SudokuPage.js

```diff
@@ -2,10 +2,12 @@ import "./SudokuPage.css";
 import SudokuGrid from "../components/SudokuGrid/SudokuGrid.js";
 import NumberInputField from "../components/NumberInputField.js";
 import ValidateBtn from "../components/ValidateBtn.js";
-import { useEffect, useState } from "react";
+import cloneMatrix from "../utility/cloneMatrix.js";
+import { useEffect, useState, useRef } from "react";
 
 export default function SudokuPage() {
-  const [numbers, setNumbers] = useState([[], [], [], [], [], [], [], [], []]);
+  const [numbers, setNumbers] = useState(Array(9).fill([]));
```

Comment:
i like that solution much better :)

Reply:
Me too. Namir gave me the tip to write it like that;)